app/models/deploy.rb

```diff
@@ -97,6 +97,16 @@ def pending_start!
     DeployService.new(user).confirm_deploy!(self)
   end
 
+  def self.in_range(time_value, time_unit)
+    where(created_at: (Time.now - time_value.public_send(time_unit))..Time.now)
```

Comment:
also can be simplified to the faster `"created_at > ?", xxx.ago` (1 time comparison vs 2)